pkg/registry/admissionregistration/mutatingadmissionpolicybinding/authz_test.go

```diff
@@ -0,0 +1,162 @@
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/kubernetes/pkg/apis/admissionregistration"
+	"k8s.io/kubernetes/pkg/registry/admissionregistration/resolver"
+)
+
+func TestAuthorization(t *testing.T) {
+	for _, tc := range []struct {
+		name             string
+		userInfo         user.Info
+		auth             AuthFunc
+		policyGetter     PolicyGetterFunc
+		resourceResolver resolver.ResourceResolverFunc
+		expectErr        bool
+	}{
+		{
+			name:      "superuser",
+			userInfo:  &user.DefaultInfo{Groups: []string{user.SystemPrivilegedGroup}},
+			expectErr: false, // success despite always-denying authorizer
+			auth: func(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+				return authorizer.DecisionDeny, "", nil
+			},
+		},
+		{
+			name:     "authorized",
+			userInfo: &user.DefaultInfo{Groups: []string{user.AllAuthenticated}},
+			auth: func(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+				if a.GetResource() == "configmaps" {
+					return authorizer.DecisionAllow, "", nil
+				}
+				return authorizer.DecisionDeny, "", nil
+			},
+			policyGetter: func(ctx context.Context, name string) (*admissionregistration.MutatingAdmissionPolicy, error) {
+				return &admissionregistration.MutatingAdmissionPolicy{
+					ObjectMeta: metav1.ObjectMeta{Name: "replicalimit-policy.example.com"},
+					Spec: admissionregistration.MutatingAdmissionPolicySpec{
+						ParamKind: &admissionregistration.ParamKind{Kind: "ConfigMap", APIVersion: "v1"},
+					},
+				}, nil
+			},
+			resourceResolver: func(gvk schema.GroupVersionKind) (schema.GroupVersionResource, error) {
+				return schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "configmaps",
+				}, nil
+			},
+			expectErr: false,
+		},
+		{
+			name:     "denied",
+			userInfo: &user.DefaultInfo{Groups: []string{user.AllAuthenticated}},
+			auth: func(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+				if a.GetResource() == "configmaps" {
+					return authorizer.DecisionAllow, "", nil
+				}
+				return authorizer.DecisionDeny, "", nil
+			},
+			policyGetter: func(ctx context.Context, name string) (*admissionregistration.MutatingAdmissionPolicy, error) {
+				return &admissionregistration.MutatingAdmissionPolicy{
+					ObjectMeta: metav1.ObjectMeta{Name: "replicalimit-policy.example.com"},
+					Spec: admissionregistration.MutatingAdmissionPolicySpec{
+						ParamKind: &admissionregistration.ParamKind{Kind: "Params", APIVersion: "foo.example.com/v1"},
+					},
+				}, nil
+			},
+			resourceResolver: func(gvk schema.GroupVersionKind) (schema.GroupVersionResource, error) {
+				return schema.GroupVersionResource{
+					Group:    "foo.example.com",
+					Version:  "v1",
+					Resource: "params",
+				}, nil
+			},
+			expectErr: true,
```

Comment:
case with NotFound on policy get is missing.

Reply:
cases missing when relevant fields are NOT updated

Reply:
I've added a 'param not found' case.  I'm not clear on what a case for relevant fields NOT updated would entail. Can you say any more about what you noticed missing there?

Reply:
I added a few other tests as well. PTAL

Reply:
I meant calling `errs := strategy.ValidateUpdate(ctx, obj, obj)` with 
```
auth: func(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
	return authorizer.DecisionDeny, "", nil
},
```
Then `errs` should be empty.

Reply:
Thanks, added.